src/fuddly/framework/plumbing.py

```diff
@@ -759,125 +758,159 @@ def monitor_probes(self, prefix=None, force_record=False):
 
     @EnforceOrder(initial_func=True, final_state="get_projs")
     def get_data_models(self, fmkDB_update=True):
+        self._get_data_models_from_fs(fmkDB_update)
+        self._get_data_models_from_modules(fmkDB_update)
+
+        if fmkDB_update:
+            self.fmkDB.insert_data_model(Database.DEFAULT_DM_NAME)
+            self.fmkDB.insert_dmaker(Database.DEFAULT_DM_NAME, Database.DEFAULT_GTYPE_NAME,
+                                     Database.DEFAULT_GEN_NAME, True, True)
+
+    def _get_data_models_from_fs(self, fmkDB_update=True):
+        if not self._quiet:
+            self.print(colorize(FontStyle.BOLD + "=" * 63 + "[ Data Models (filesystem) ]==", rgb=Color.FMKINFOGROUP))
+
         data_models = collections.OrderedDict()
 
         def populate_data_models(path, prefix=""):
-            dm_dir = os.path.basename(os.path.normpath(path))
-            for (dirpath, dirnames, filenames) in os.walk(path):
-                if filenames:
-                    data_models[prefix + dm_dir] = []
-                    data_models[prefix + dm_dir].extend(filenames)
-                for d in dirnames:
-                    full_path = os.path.join(path, d)
-                    rel_path = os.path.join(dm_dir, d)
-                    data_models[prefix + rel_path] = []
-                    for (dth, dnames, fnm) in os.walk(full_path):
-                        data_models[prefix + rel_path].extend(fnm)
-                        break
-                break
-
-        populate_data_models(gr.data_models_folder, prefix="fuddly/")
+            from pathlib import Path
+            from os.path import dirname,basename
+
+            if path[-1] == '/':
+                path = path[:-1]
+            if prefix != "" and prefix[-1] != '/':
+                prefix = prefix + '/'
+
+            base_path = dirname(path)
+            _, dirs, _ = next(os.walk(path))
+            for dirpath in dirs:
+                p = Path(os.path.join(path, dirpath))
+                inits = [ dirname(x) for x in  p.glob('**/__init__.py') ]
+                strats = [ dirname(x) for x in  p.glob('**/strategy.py') ]
+                modules = list(set(inits) & set(strats))
+                for m in modules:
+                    relpath = dirname(m)[len(base_path)+1:]
+                    key=prefix+relpath
+                    if data_models.get(key) is None:
```

Comment:
use defaultdict ?

Reply:
For what, data_model ? It is a `colloections.OrderedDict()` (based on what was used before)